process/economics/economicsData.go

```diff
@@ -191,10 +208,15 @@ func (ed *EconomicsData) MaxGasLimitPerBlock() uint64 {
 func (ed *EconomicsData) ComputeGasLimit(tx process.TransactionWithFeeHandler) uint64 {
 	gasLimit := ed.minGasLimit
 
-	//TODO: change this method of computing the gas limit of a notarizing tx
-	// it should follow an exponential curve as to disincentivise notarizing large data
-	// also, take into account if destination address is 0000...00000 as this will be a SC deploy tx
-	gasLimit += uint64(len(tx.GetData()))
+	dataLen := uint64(len(tx.GetData()))
+	gasLimit += dataLen * ed.gasPerDataByte
+
+	if dataLen < ed.dataLimitForBaseCalc || core.IsEmptyAddress(tx.GetRecvAddress()) {
+		return gasLimit
+	}
+
+	overDataLimit := dataLen - ed.dataLimitForBaseCalc
```

Comment:
suggestion to leave gasLimit to grow in a linearly fashion. so lines 214:219 can be commented out with a TODO for re-evaluation or removed.

Reply:
done.